python/ray/_private/runtime_env/agent/main.py

```diff
@@ -0,0 +1,197 @@
+    )
+    parser.add_argument(
+        "--logging-rotate-backup-count",
+        required=False,
+        type=int,
+        default=ray_constants.LOGGING_ROTATE_BACKUP_COUNT,
+        help="Specify the backup count of rotated log file, default is {}.".format(
+            ray_constants.LOGGING_ROTATE_BACKUP_COUNT
+        ),
+    )
+    parser.add_argument(
+        "--log-dir",
+        required=True,
+        type=str,
+        default=None,
+        help="Specify the path of log directory.",
+    )
+    parser.add_argument(
+        "--temp-dir",
+        required=True,
+        type=str,
+        default=None,
+        help="Specify the path of the temporary directory use by Ray process.",
+    )
+
+    args = parser.parse_args()
+
+    logging_params = dict(
+        logging_level=args.logging_level,
+        logging_format=args.logging_format,
+        log_dir=args.log_dir,
+        filename=args.logging_filename,
+        max_bytes=args.logging_rotate_bytes,
+        backup_count=args.logging_rotate_backup_count,
+    )
+
+    # Setup stdout/stderr redirect files
+    out_file, err_file = open_capture_files(args.log_dir)
+    configure_log_file(out_file, err_file)
+
+    agent = RuntimeEnvAgent(
+        runtime_env_dir=args.runtime_env_dir,
+        logging_params=logging_params,
+        gcs_address=args.gcs_address,
+        temp_dir=args.temp_dir,
+        address=args.node_ip_address,
+        runtime_env_agent_port=args.runtime_env_agent_port,
+    )
+
+    # POST /get_or_create_runtime_env
+    # body is serialzied protobuf GetOrCreateRuntimeEnvRequest
+    # reply is serialzied protobuf GetOrCreateRuntimeEnvReply
+    async def get_or_create_runtime_env(request):
+        data = await request.read()
+        request = runtime_env_agent_pb2.GetOrCreateRuntimeEnvRequest()
+        request.ParseFromString(data)
+        reply = await agent.GetOrCreateRuntimeEnv(request)
+        return web.Response(
+            body=reply.SerializeToString(), content_type="application/octet-stream"
+        )
+
+    # POST /delete_runtime_env_if_possible
+    # body is serialzied protobuf DeleteRuntimeEnvIfPossibleRequest
+    # reply is serialzied protobuf DeleteRuntimeEnvIfPossibleReply
+    async def delete_runtime_env_if_possible(request):
+        data = await request.read()
+        request = runtime_env_agent_pb2.DeleteRuntimeEnvIfPossibleRequest()
+        request.ParseFromString(data)
+        reply = await agent.DeleteRuntimeEnvIfPossible(request)
+        return web.Response(
+            body=reply.SerializeToString(), content_type="application/octet-stream"
+        )
+
+    # POST /get_runtime_envs_info
+    # body is serialzied protobuf GetRuntimeEnvsInfoRequest
+    # reply is serialzied protobuf GetRuntimeEnvsInfoReply
+    async def get_runtime_envs_info(request):
+        data = await request.read()
+        request = runtime_env_agent_pb2.GetRuntimeEnvsInfoRequest()
```

Comment:
Q: if the parsing fails here, what's going to happen? Is there any good way to do validation?

Reply:
it raises (protobuf's) message.DecodeError. aiohttp handles this by returning a 500:

```
< HTTP/1.1 500 Internal Server Error
< Content-Type: text/plain; charset=utf-8
< Content-Length: 55
< Date: Fri, 07 Jul 2023 18:40:54 GMT
< Server: Python/3.9 aiohttp/3.8.4
< Connection: close
<
500 Internal Server Error

* Closing connection 0
Server got itself in trouble%
```